### PR TITLE
Add post on "Making popular Ruby packages more secure" (MFA Phase 2 Rollout)

### LINKED ISSUE #109: Create a blog post to announce MFA policy changes
Contributes to [MFA RFC](https://github.com/rubygems/rfcs/pull/36/files#diff-3d5cc3acc06fe7e9150fdbfc43399c5ad42572c122187774bfc3a4857df524f1R46-R67)

We will write a blog post to announce the new MFA policy changes. The initial announcement will outline [Phase 2's](https://github.com/rubygems/rfcs/pull/36/files#diff-3d5cc3acc06fe7e9150fdbfc43399c5ad42572c122187774bfc3a4857df524f1R46-R67) plan – recommending MFA for Top-100 Rubygems package maintainers.

As part of the communications effort, @sonalkr132 [has also recommended](https://bundler.slack.com/archives/C0HJMKWH4/p1648654978626329?thread_ts=1648491986.485729&cid=C0HJMKWH4) that we share the announcement through other channels.

### 📝 To Do's
- [x] Write copy for blog post
- [ ] Publish blog post
- [ ] Share post on `Ruby Weekly`
- [ ] Share post on `reddit` (e.g. `r/ruby`)

### PR DESCRIPTION
**Pull request by bettymakes**
Closes #109

### 🎯 Why do we need this post?
This post is part of the communications plan to announce the MFA policy changes that will be rolled out to the top-100 gem maintainers.

### 🗓 When should we make this announcement?
Once all [the technical requirements](https://github.com/orgs/rubygems/projects/3) have been implemented, reviewed, and are ready to be merged. We can synchronize this post with the merging of the code changes. 

[Update: 2022/06/06] We met with @sonalkr132 & @simi and aligned on announcing this on June 13, 2022.

### 🤔 What feedback are we looking for in reviewing this announcement?
**There are a number of items I'm uncertain about:**

1️⃣ Who should be tagged as the author of this post? I'm thinking it should be someone from the RubyGems team?
  * ✅ @sonalkr132 addressed in [comment](https://github.com/rubygems/rubygems.github.io/pull/110#discussion_r867524072)

2️⃣ Are you happy with the title? Would you prefer to call out `Top-100 RubyGems package maintainers` explicitly? My reservation was that it felt like too much of a mouthful for a title.
  * ✅ No longer relevant concern. We renamed to `Making popular Ruby packages more secure` 

3️⃣ Would you like to reference back to the [RFC](https://github.com/rubygems/rfcs/pull/36) in this post?
  * ✅ @sonalkr132 addressed in [comment](https://github.com/rubygems/rubygems.github.io/pull/110#issuecomment-1120464708)

### CODE REVIEW

File: _posts/2022-04-08-mfa-phase-1-rollout.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: Making popular Ruby packages more secure
+layout: post
+author: 
+author_email:
```
Comment (bettymakes) [inline _posts/2022-04-08-mfa-phase-1-rollout.md:5]: Not sure who should be the author. Thinking it should be someone from the RubyGems Team?

File: _posts/2022-04-08-mfa-phase-1-rollout.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+title: Making popular Ruby packages more secure
+layout: post
+author: 
+author_email: 
+---
+
+Attacks on the software supply chain are increasing. Our community has not gone unscathed. RubyGems has been affected by supply chain attacks in the past. It’s important for us to mitigate these risks as much as possible. Recommending stronger security practices like enabling multi-factor authentication (MFA) on high-impact packages will be a first step in improving security of the RubyGems ecosystem.
```
Comment (bettymakes) [inline _posts/2022-04-08-mfa-phase-1-rollout.md:8]: > Our community has not gone unscathed. RubyGems has been affected by supply chain attacks in the past.

I'm writing this with the best of intentions ❤️ . The rationale behind these sentences was to drive a compelling reason why RubyGems is doubling down on security. But I'm also cognizant that the Team may not want to call further attention to past incidents. If y'all want to drop this or replace it, I'm fully supportive of that ☺️.

File: _posts/2022-04-08-mfa-phase-1-rollout.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Account takeovers are the second most common attack on software supply chains ([source](https://arxiv.org/abs/2002.01139)). The countermeasure against this type of attack is simple: enabling MFA. Doing so will help reduce the chances of a successful attack by 99.9 percent ([source](https://www.microsoft.com/security/blog/2019/08/20/one-simple-action-you-can-take-to-prevent-99-9-percent-of-account-attacks/)).
+
+Starting today, top-100 RubyGems package maintainers will begin to see reminders on the RubyGems command-line tool and website if MFA is not enabled on their accounts. Anybody who maintains a gem with more than 165 million downloads will see this recommendation.
+
+Although this is currently just a recommendation, we will enforce MFA on high-impact gems in the future. This policy will bring us in line with other package ecosystems (e.g. [npm](https://github.blog/2022-02-01-top-100-npm-package-maintainers-require-2fa-additional-security/)).
```
Comment (bettymakes) [inline _posts/2022-04-08-mfa-phase-1-rollout.md:14]: Terminology question:
I noticed NPM referred to the top-100 packages as "high-impact". I liked that terminology and have used it for us too. Do you like "high-impact"? Or would you prefer using `top-100 gems` or `top-100 packages` instead?

File: _posts/2022-04-08-mfa-phase-1-rollout.md (hunk 1 above)
Comment (sonalkr132) [reply _posts/2022-04-08-mfa-phase-1-rollout.md:5]: You can be the author. If you prefer, you can add your team name as well.

File: _posts/2022-04-08-mfa-phase-1-rollout.md (hunk 2 above)
Comment (sonalkr132) [reply _posts/2022-04-08-mfa-phase-1-rollout.md:8]: > Recommending stronger security practices like enabling multi-factor authentication (MFA) on high-impact packages will be a first step in improving security of the RubyGems ecosystem.

"will be" seems out of place. perhaps "is a first step"?

File: _posts/2022-04-08-mfa-phase-1-rollout.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+layout: post
+author: 
+author_email: 
+---
+
+Attacks on the software supply chain are increasing. Our community has not gone unscathed. RubyGems has been affected by supply chain attacks in the past. It’s important for us to mitigate these risks as much as possible. Recommending stronger security practices like enabling multi-factor authentication (MFA) on high-impact packages will be a first step in improving security of the RubyGems ecosystem.
+
+Account takeovers are the second most common attack on software supply chains ([source](https://arxiv.org/abs/2002.01139)). The countermeasure against this type of attack is simple: enabling MFA. Doing so will help reduce the chances of a successful attack by 99.9 percent ([source](https://www.microsoft.com/security/blog/2019/08/20/one-simple-action-you-can-take-to-prevent-99-9-percent-of-account-attacks/)).
+
+Starting today, top-100 RubyGems package maintainers will begin to see reminders on the RubyGems command-line tool and website if MFA is not enabled on their accounts. Anybody who maintains a gem with more than 165 million downloads will see this recommendation.
```
Comment (sonalkr132) [inline _posts/2022-04-08-mfa-phase-1-rollout.md:12]: Adding date and month may be more precise instead of "today".

Comment (sonalkr132) [reply _posts/2022-04-08-mfa-phase-1-rollout.md:12]: top-100 and 165 mil+ downloads may not always be accurate. lets talk about download count first and then clarify that we are hoping to target *at least* top 100 gems by downloads.

File: _posts/2022-04-08-mfa-phase-1-rollout.md (hunk 3 above)
Comment (sonalkr132) [reply _posts/2022-04-08-mfa-phase-1-rollout.md:14]: top 100 is fine. impact wording is questionable given that transitive dependencies are not covered.

**Comment by sonalkr132** [comment]
I would also link the RFC and add a dedicated section about future plans. Future plans could be that we are having a discussion about increasing the coverage here | link and these are current ideas (transitive vs full coverage). Perhaps we can also suggest that readers can use rubygems.org issue page or slack if they have any thoughts about the plan of action.

**Comment by tomstuart** [comment]
Apologies, I prematurely pushed some speculative edits and have since reset the branch. I’ll add them back shortly.

**Comment by bettymakes** [comment]
👋 Hey @sonalkr132! We've revised the post based on your feedback. It should be ready for another review 👀.

**Comment by bettymakes** [comment]
I'm not sure if it's too whimsical or off-brand for the blog, but we drew a little doodle of a MFA Gem. 
@aellispierce & I created some doodles for our talk at RailsConf and gave out some MFA Gem stickers.

Happy to include the doodle in the blog (and/or email) if we're open to adding some extra ✨ flare ✨ to the blog post.
Totally ok if you think it's not on-brand with RubyGems though 👌.

<p align="center">
  <kbd>
    Gemmy wearing an MFA hat, holding a checked flag.
  </kbd>
</p>

File: _posts/2022-06-13-mfa-phase-1-rollout.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+As proposed in the [MFA rollout RFC](https://github.com/rubygems/rfcs/blob/master/text/0007-mfa-rollout.md), RubyGems is beginning to move towards greater adoption of multi-factor authentication for gem maintainers.
+
+Starting today (June 13, 2022), the maintainers of at least the top 100 RubyGems packages will begin to see warnings on the RubyGems command-line tool and website if MFA is not enabled on their accounts. Anybody who maintains a gem with more than 165 million downloads will see this recommendation.
+
+Although this is currently just a recommendation, we will begin to enforce MFA for maintainers of those gems in two months’ time (August 15, 2022). This policy will bring us in line with other package ecosystems (e.g. [npm](https://github.blog/2022-02-01-top-100-npm-package-maintainers-require-2fa-additional-security/)) as well as [GitHub](https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/).
```
Comment (sonalkr132) [inline _posts/2022-06-13-mfa-phase-1-rollout.md:16]: Lets also talk about email reminders to clarify who is affected. Something along the lines:
> If you are owner of any of these gems, you will receive an email reminder to enable mfa one month and one week before this goes into effect.

We should clarify that MFA requirement means ui_and_gem_signin only. Link the guide to clarify what the level means. We can also add that we recommend enableing ui_and_api level if possible.

File: _posts/2022-06-13-mfa-phase-1-rollout.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+We are committed to fostering a safer, more secure ecosystem for Rubyists. This is one of many steps we plan to take over the coming months in order to maintain a healthy, reliable ecosystem for everyone. Stay tuned!
```
Comment (sonalkr132) [inline _posts/2022-06-13-mfa-phase-1-rollout.md:18]: It is upto you but if you want, you and mention that we plan to increase the MFA coverage by extending this requirement to more gems in future. If the reader has opinons about this, they can reach us in slack or github issues.

**Comment by sonalkr132** [comment]
> Happy to include the doodle in the blog (and/or email) if we're open to adding some extra sparkles flare sparkles to the blog post.

Please feel free.

**Review by sonalkr132 — CHANGES_REQUESTED** [review]
needs more changes.

**Comment by bettymakes** [comment]
Hey Aditya 👋 ! I've just [merged in revisions](https://github.com/Shopify/rubygems.github.io/pull/2) based on your latest feedback. Let me know if there's anything else.

**Review by tomstuart — COMMENTED** [review]
Thank you Betty, I think this is good to go! 👍🏻

**Review by dwradcliffe — APPROVED** [review]
LGTM, thanks!

**Review by sonalkr132 — APPROVED** [review]

**Pull request merged**